Maintained by Andrew <andrew@webupd8.org>
Index: nemo-1.7.6/data/nemo.desktop.in.in
===================================================================
--- nemo-1.7.6.orig/data/nemo.desktop.in.in	2013-05-04 10:02:46.885816391 +1000
+++ nemo-1.7.6/data/nemo.desktop.in.in	2013-05-04 10:02:46.882483057 +1000
@@ -276,7 +276,7 @@
 Comment[zh_TW]=存取與組織檔案
 Comment[zu]=Access and organize files
 Exec=nemo %U
-Icon=folder
+Icon=system-file-manager
 Terminal=false
 Type=Application
 StartupNotify=false
