Maintained by Andrew <andrew@webupd8.org>
Index: nemo-1.7.6/data/nemo.desktop.in.in
#  Fix bad positioning of rename window in some views (gtk 3.10) : https://github.com/linuxmint/nemo/commit/d5377cfd53034b6ab5588376322126f48782ec59
===================================================================
--- nemo-2.0.8.orig/eel/eel-canvas.c 	2013-05-04 10:02:46.885816391 +1000
+++ nemo-2.0.8/eel/eel-canvas.c 	2013-05-04 10:02:46.882483057 +1000
@@ -3213,11 +3213,20 @@
                 return FALSE;
 
         bin_window = gtk_layout_get_bin_window (GTK_LAYOUT (widget));
+
+        if (!gtk_cairo_should_draw_window (cr, bin_window))
+            return FALSE;
+
+        cairo_save (cr);
+
         gtk_cairo_transform_to_window (cr, widget, bin_window);
 
         region = eel_cairo_get_clip_region (cr);
-        if (region == NULL)
-                return FALSE;
+
+        if (region == NULL) {
+            cairo_restore (cr);
+            return FALSE;
+        }
 
 #ifdef VERBOSE
 	g_print ("Draw\n");
@@ -3248,6 +3257,8 @@
 	
 	if (canvas->root->flags & EEL_CANVAS_ITEM_MAPPED)
 		EEL_CANVAS_ITEM_GET_CLASS (canvas->root)->draw (canvas->root, cr, region);
+
+    cairo_restore (cr);
 
 	/* Chain up to get exposes on child widgets */
         if (GTK_WIDGET_CLASS (canvas_parent_class)->draw)

