Maintained by Andrew <andrew@webupd8.org>
Index: nemo-1.7.6/data/nemo.desktop.in.in
#  Stop GTK_STOCK deprecated errors for gtk-3.10
===================================================================
--- nemo-2.0.8.orig/configure.in 	2013-05-04 10:02:46.885816391 +1000
+++ nemo-2.0.8/configure.in	2013-05-04 10:02:46.882483057 +1000
@@ -296,7 +296,6 @@
 AC_SUBST(NEMO_CFLAGS)
 AC_SUBST(NEMO_LIBS)
 
-DISABLE_DEPRECATED_CFLAGS="-DG_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED -DGDK_PIXBUF_DISABLE_DEPRECATED"
 AC_SUBST(DISABLE_DEPRECATED_CFLAGS)
 
 dnl Multimedia keys

