Maintained by Andrew <andrew@webupd8.org>
Index: nemo/src/nemo-icon-view-container.c
# Fix hidpi-mode crash when using icon emblems on odd-shaped thumbnails https://github.com/linuxmint/nemo/commit/1a94c498ff9e193934ca67f31ee075b6a26c397d
===================================================================
--- nemo-2.2.0.orig/src/nemo-icon-view-container.c 	2014-05-04 10:02:46.885816391 +1000
+++ nemo-2.2.0/src/nemo-icon-view-container.c	2014-05-04 10:02:46.882483057 +1000
@@ -126,8 +126,8 @@ nemo_icon_view_container_get_icon_images (NemoIconContainer *container,
         if (s < size)
             size = s;

-        bad_ratio = nemo_icon_get_emblem_size_for_icon_size (size) > w ||
-                    nemo_icon_get_emblem_size_for_icon_size (size) > h;
+        bad_ratio = nemo_icon_get_emblem_size_for_icon_size (size) * scale > w ||
+                    nemo_icon_get_emblem_size_for_icon_size (size) * scale > h;

         if (bad_ratio)
             goto skip_emblem; /* Would prefer to not use goto, but

--- nemo-2.2.0.orig/src/nemo-list-model.c 	2014-05-04 10:02:46.885816392 +1000
+++ nemo-2.2.0/src/nemo-list-model.c	2014-05-04 10:02:46.882483058 +1000
@@ -344,8 +344,8 @@ nemo_list_model_get_value (GtkTreeModel *tree_model, GtkTreeIter *iter, int colu
             if (s < icon_size)
                 icon_size = s;

-            bad_ratio = nemo_icon_get_emblem_size_for_icon_size (icon_size) > w ||
-                        nemo_icon_get_emblem_size_for_icon_size (icon_size) > h;
+            bad_ratio = nemo_icon_get_emblem_size_for_icon_size (icon_size) * icon_scale > w ||
+                        nemo_icon_get_emblem_size_for_icon_size (icon_size) * icon_scale > h;

 			gicon = G_ICON (pixbuf);

--- nemo-2.2.0.orig/libnemo-private/nemo-icon-info.c 	2014-05-04 10:02:46.885816393 +1000
+++ nemo-2.2.0/libnemo-private/nemo-icon-info.c   	2014-05-04 10:02:46.882483059 +1000
@@ -553,7 +553,7 @@ nemo_icon_info_get_pixbuf_nodefault_at_size (NemoIconInfo  *icon,

 	scale = (double)forced_size / s;
 	scaled_pixbuf = gdk_pixbuf_scale_simple (pixbuf,
-						 w * scale, h * scale,
+						 MAX (w * scale, 1), MAX (h * scale, 1),
 						 GDK_INTERP_BILINEAR);
 	g_object_unref (pixbuf);
 	return scaled_pixbuf;
@@ -579,7 +579,7 @@ nemo_icon_info_get_pixbuf_at_size (NemoIconInfo  *icon,

 	scale = (double)forced_size / s;
 	scaled_pixbuf = gdk_pixbuf_scale_simple (pixbuf,
-						 w * scale, h * scale,
+						 MAX (w * scale, 1), MAX (h * scale, 1),
 						 GDK_INTERP_BILINEAR);
 	g_object_unref (pixbuf);
 	return scaled_pixbuf;

