From b3b32f4afb5c1a5412b5fed80c8a0ed2ddef6e65 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Mon, 17 Feb 2014 22:13:52 +0000
Subject: slot: don't set spacing on extra location widgets vbox

https://bugzilla.gnome.org/show_bug.cgi?id=721647
---
diff --git a/src/nemo-window-slot.c b/src/nemo-window-slot.c
index 1d38503..a66d4b0 100644
--- a/src/nemo-window-slot.c
+++ b/src/nemo-window-slot.c
@@ -178,7 +178,7 @@
 					GTK_ORIENTATION_VERTICAL);
 	gtk_widget_show (GTK_WIDGET (slot));
 
-	extras_vbox = gtk_box_new (GTK_ORIENTATION_VERTICAL, 6);
+	extras_vbox = gtk_box_new (GTK_ORIENTATION_VERTICAL, 0);
 	slot->extra_location_widgets = extras_vbox;
 	gtk_box_pack_start (GTK_BOX (slot), extras_vbox, FALSE, FALSE, 0);
 	gtk_widget_show (extras_vbox);
--
cgit v0.9.2

